lambdex/bin/lambdex.py

```diff
@@ -60,6 +61,7 @@ def write_lambdex_handler(pex_zip, options):
     if options.output:
         output_zip = options.output
         shutil.copy(pex_zip, output_zip)
+        os.chmod(output_zip, os.stat(output_zip).st_mode | stat.S_IWRITE)
```

Comment:
Is this a shot in the dark? shutil.copy is supposed to preserve perms. If that's true that implies the else branch below produces an un-writable zip since you have to add write perms here. That means Lambdex now behaves differently if you specify --output above and beyond the expected difference.

Reply:
I remember checking that `shutil.copy` was supposed to clear perms on the original review... it's really odd that this is necessary.

Reply:
...oh. But it totally doesn't if you play with it. That's annoying.
```python
$ python3
>>> shutil.copy("file.txt", "file.2.txt")
$ ls -l
total 0
-r--r-----  1 stuhood  staff  0 Feb 14 20:33 file.2.txt
-r--r-----  1 stuhood  staff  0 Feb 14 20:32 file.txt
```

Reply:
@jsirois : It seems reasonable to me that if asked to create a new file a umask would (effectively) be applied, but that if given an existing file, permissions would not be adjusted.

Reply:
> I remember checking that `shutil.copy` was supposed to clear perms on the original review... it's really odd that this is necessary.

Oh, jeez: I looked at `shutil.copyfile`, which explicitly does not preserve permissions: https://docs.python.org/3/library/shutil.html#shutil.copyfile

Consider using that instead?

Reply:
@stuhood that might be true, but the acid test here is pretending we don't own Lambdex. We'd work around this in the Pants rule in that case.

Reply:
My concern is that the remexec sandboxed input files will not have the write bit set to begin with. In which case preserving permissions is not a solution.

Reply:
Got it. I do want to re-stress that the mere mention of remexec here is a red flag, this is a Pants problem. I'll desist though.

Reply:
I don't think changing permissions in both branches is the right thing to do. If I provide Lambdex with a file to write in-place, it's my responsibility to ensure that the file is writeable. That seems very reasonable. OTOH If I ask Lambdex to output to a different file, then the fact that it uses shutil.copy() is an implementation detail that happens to require us to reason about the write bit internally.

E.g., we could get rid of shutil.copy and instead copy the bytes to the new location in a loop, and evade the problem entirely.

Reply:
And this has nothing to do with remexec or Pants. If I specify an output file then surely I am under no obligation to ensure that the *input* file is writeable. That would not be at all intuitive.

Reply:
I don't see this as a Pants problem.See above for why.